src/Backend/ONE/ONEToolchain.ts

```diff
@@ -14,33 +14,106 @@
+  private readonly debianArch: DebianArch;
 
   constructor() {
-    this.toolchainTypes = [];
+    this.toolchainName = 'onecc-docker-test';
+    // we currently support only the latest version of one-compiler.
+    this.toolchainTypes = ['latest'];
+
+    // we temporary add personal PPA for test.
+    this.debianRepo =
+        new DebianRepo('http://ppa.launchpad.net/psalmist-kim/ppa/ubuntu', 'bionic', 'main');
+    this.debianArch = DebianArch.amd64;
   }
 
   getToolchainTypes(): string[] {
     return this.toolchainTypes;
   }
 
-  getToolchains(_toolchainType: string, _start: number, _count: number): Toolchains {
-    throw Error('Invalid getToolchains call');
+  getToolchains(_toolchainType: string, _start?: number, _count?: number): Toolchains {
+    if (parseInt(execSync(`apt-cache show ${this.toolchainName} > /dev/null 2>&1; echo $?`)
+                     .toString()
+                     .trim()) !== 0) {
+      throw Error('the prerequisites are not met');
+    }
+
+    const availableToolchains = new Toolchains();
+
+    if (_toolchainType === 'latest') {
+      const versionList =
+          execSync(`apt-cache show ${this.toolchainName} | grep Version | awk '{print $2}'`)
+              .toString()
+              .trim()
+              .split('\n');
+
+      for (const version of versionList) {
+        const [major, minor, patch, option] = version.split(/[.~]+/);
+
+        const toolchainInfo = new ToolchainInfo(
+            this.toolchainName, 'Description: test',
+            new Version(parseInt(major), parseInt(minor), parseInt(patch), option));
+
+        const toolchain = new DebianToolchain(toolchainInfo, this.debianRepo, this.debianArch);
+        availableToolchains.push(toolchain);
+      }
+    }
+
+    return availableToolchains;
   }
 
   getInstalledToolchains(_toolchainType: string): Toolchains {
-    throw Error('Invalid getInstalledToolchains call');
+    const availableToolchains = new Toolchains();
+
+    if (_toolchainType === 'latest') {
+      const versionList =
+          execSync(`apt-cache show ${this.toolchainName} | grep Version | awk '{print $2}'`)
+              .toString()
+              .trim()
+              .split('\n');
+
+      for (const version of versionList) {
+        const [major, minor, patch, option] = version.split(/[.~]+/);
+
+        const toolchainInfo = new ToolchainInfo(
+            this.toolchainName, 'Description: test',
+            new Version(parseInt(major), parseInt(minor), parseInt(patch), option));
+
+        const toolchain = new DebianToolchain(toolchainInfo, this.debianRepo, this.debianArch);
+        availableToolchains.push(toolchain);
+      }
+    }
+
+    const installedToolchains = availableToolchains.filter((toolchain) => {
+      return parseInt(execSync(`dpkg-query --show ${toolchain.info.name} > /dev/null 2>&1; echo $?`)
+                          .toString()
+                          .trim()) === 0;
+    });
+    return installedToolchains;
```

Comment:
Um, do you need `availableToolchains` list? I think the `dpkg-query --show <toolchain name>` command is enough to get an installed package list.

Reply:
It seems unnecessary. I thought there was a possibility of having multiple toolchains installed, but I think I'm wrong.